plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
+		if checkIntegerOverFlow(totalPriority, config.Weight*MaxPriority) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * MaxPriority
+	}
 	return configs, nil
 }
 
+// checkIntegerOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkIntegerOverFlow(totalPriority int, weight int) bool {
```

Comment:
@jayunit100 , @k82cn, I am fine with making those changes but AFAIU, it seems weight can never overflow, even if it overflows, shouldn't we check them in functions where they were getting updated?